presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -536,15 +536,20 @@ protected ObjectListing computeNext(ObjectListing previous)
      * This exception is for stopping retries for S3 calls that shouldn't be retried.
      * For example, "Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: Forbidden (Service: Amazon S3; Status Code: 403 ..."
      */
-    @VisibleForTesting
-    static class UnrecoverableS3OperationException
+    public static class UnrecoverableS3OperationException
```

Comment:
I think we can make `UnrecoverableS3OperationException` a top level class and move it out of `PrestoS3FileSystem` as it is being used from many places right now.